common/src/terraingen.rs

```diff
@@ -0,0 +1,996 @@
+use crate::world::Material;
+
+pub struct VoronoiInfo {
+    pub location: na::Vector2<f64>,
```

Comment:
I'd be inclined to use f32 here to keep the tables smaller unless you specifically need high precision for some reason.

Reply:
I used f64 because the envirofactors are f64. I've made the struct definition of location use [f32; 2] now, converting using `as` when f64 values are passed in. Perhaps there's an argument for the envirofactors themselves being f32 instead of f64---I don't see why they need to be particularly precise, and the extra memory used adds up across the thousands of nodes.

Reply:
The environment factors were 64-bit because we do nothing to ensure they don't grow without bound, basically. That might not be a huge issue if it can be shown that it would take an unreasonable amount of travel to observe any issues, though.